### PR TITLE
Fixes #57: `attr-for-selected` now always expects hyphenated names.

### LINKED ISSUE #57: The value of attr-for-selected must be camel-cased, not hyphenated?
While working with a ``paper-dropdown-menu``, ``paper-menu``, and ``paper-item``, I couldn't understand why some of the dropdowns weren't displaying the initially selected item or allowing me to change the selection.  I finally discovered through lots of trial and error that ``attr-for-selected`` expects a camelcase value.  Declaring ``attr-for-selected="foo-bar"`` breaks things as described above.  Changing that to ``attr-for-selected="fooBar"`` makes things work as expected.  I suggest that iron-selector be fixed to allow things to work either way, or update the documentation to make it clear that camelcase is required.

### PR DESCRIPTION
**Pull request by bicknellr**
(fixes #57)

### CODE REVIEW

File: iron-selectable.html [hunk 1]
```diff
@@ -264,7 +267,8 @@
     },
 
     _valueForItem: function(item) {
-      return item[this.attrForSelected] || item.getAttribute(this.attrForSelected);
+      return item[Polymer.CaseMap.dashToCamelCase(this.attrForSelected)] ||
```
Comment (arthurevans) [inline iron-selectable.html]: Would it be worth memoizing the camelCase version when attrForSelected changes (in _updateSelected)?

File: test/attr-for-selected.html [hunk 2]
```diff
@@ -0,0 +1,153 @@
+<html>
```
Comment (bicknellr) [inline test/attr-for-selected.html]: Add copyright comment.

File: iron-selectable.html (hunk 1 above)
Comment (bicknellr) [inline iron-selectable.html]: Yeah, this is going to repeat a lot of work otherwise; fix coming in just a moment.

**Comment by mgiuffrida** [comment]
Can you be sure to mention this in the docs and release notes?

In JS, should `attrForSelected` be set to a hyphenated-attribute-name or camelCaseProperty?

**Comment by bicknellr** [comment]
@mgiuffrida This PR has a doc update [here](https://github.com/PolymerElements/iron-selector/pull/101/files#diff-c64eedc3daead72759c9014ab0eb08c4R49) - let me know if it needs any clarification. Also, the expected input is the same (hyphenated) whether you're setting it as an inline attribute or by the setter. (This PR wasn't actively implemented to support that; Polymer, to my understanding, just doesn't treat properties with `type: String` differently when they're set as inline attributes vs through the setter.)

**Comment by mgiuffrida** [comment]
:+1:

File: test/attr-for-selected.html (hunk 2 above)
Comment (bicknellr) [inline test/attr-for-selected.html]: (done)

File: iron-selectable.html [hunk 3]
```diff
@@ -280,10 +283,24 @@
...
+        var attrForSelected = this.attrForSelected;
+        var propertyNameForSelected = lastPropertyNameForSelected;
+
+        if (lastAttrForSelected !== attrForSelected) {
+          lastAttrForSelected = attrForSelected;
+          lastPropertyNameForSelected = propertyNameForSelected =
+            Polymer.CaseMap.dashToCamelCase(attrForSelected);
+        }
+
+        var propValue = item[Polymer.CaseMap.dashToCamelCase(this.attrForSelected)];
```
Comment (bicknellr) [inline iron-selectable.html]: Oops, I resolved the rebase conflicts incorrectly...

File: iron-selectable.html [hunk 4]
```diff
@@ -280,10 +283,24 @@
       }
     },
 
-    _valueForItem: function(item) {
-      var propValue = item[this.attrForSelected];
-      return propValue != undefined ? propValue : item.getAttribute(this.attrForSelected);
-    },
+    _valueForItem: (function() {
```
Comment (bicknellr) [inline iron-selectable.html]: This caching is *way* broken (static)! Don't merge until I've fixed it!

**Comment by bicknellr** [comment]
Now I know if you force push a branch while the PR is *closed* you can't reopen...
imgres-1

**Comment by bicknellr** [comment]
(reopened as #109)

**Pull request closed without merging**